app/src/main/java/io/github/gsantner/memetastic/activity/MainActivity.java

```diff
@@ -383,8 +388,35 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                     String picturePath = cursor.getString(columnIndex);
                     cursor.close();
 
-                    // String picturePath contains the path of selected Image
-                    onImageTemplateWasChosen(picturePath);
+                    if (picturePath == null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+                        // Retrieve image from Cloud, e.g.: Google Drive, Picasa
+                        try {
+                            ParcelFileDescriptor parcelFileDescriptor = getContentResolver().openFileDescriptor(selectedImage, "r");
+                            if (parcelFileDescriptor != null) {
+                                FileDescriptor fileDescriptor = parcelFileDescriptor.getFileDescriptor();
+                                FileInputStream input = new FileInputStream(fileDescriptor);
+
+                                // Create temporary file in cache directory
+                                picturePath = File.createTempFile("image", "tmp", getCacheDir()).getAbsolutePath();
+                                FileOutputStream output = new FileOutputStream(picturePath);
+
+                                int read = 0;
```

Comment:
Theres already a FileUtils methods for exactly this purpose

Reply:
Ah, I didn't notice this project already includes `FileUtils`. I'll fix it.

Reply:
yep, but not the apache one, the opoc one :)